### PR TITLE
CM-203 Implementar validacion de usuarios

### PR DESCRIPTION
**Pull request by merlindk**

### CODE REVIEW

**Comment by merlindk** [comment]
test this please

File: src/main/java/com/cema/users/controllers/UserController.java [hunk 1]
```diff
@@ -48,11 +49,45 @@
...
+            @PathVariable("username") String userName) {
+
+        userName = userName.toLowerCase();
+        LOG.info("Request for user {}", userName);
+
+        CemaUser cemaUser = cemaUserRepository.findCemaUserByUserName(userName);
+        if (cemaUser == null) {
+            throw new NotFoundException(String.format(Messages.USER_DOES_NOT_EXISTS, userName));
+        }
+        if (!authorizationService.isOnTheSameEstablishment(cemaUser.getEstablishmentCuig())) {
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/controllers/UserController.java:83]: shouldn't here be an admin validation?

File: src/main/java/com/cema/users/controllers/UserController.java [hunk 2]
```diff
@@ -48,11 +49,45 @@
...
-    public UserController(CemaUserRepository cemaUserRepository, UserMapping userMapping, AuthorizationService authorizationService) {
+    public UserController(CemaUserRepository cemaUserRepository, UserMapping userMapping,
+                          AuthorizationService authorizationService, UserValidationService userValidationService) {
         this.cemaUserRepository = cemaUserRepository;
         this.userMapping = userMapping;
         this.authorizationService = authorizationService;
+        this.userValidationService = userValidationService;
+    }
+
+    @ApiOperation(value = "Retrieve a user data by username", response = User.class)
```
Comment (FranciscodeMaussion) [inline src/main/java/com/cema/users/controllers/UserController.java]: is this the real description of the method?

File: src/main/java/com/cema/users/controllers/UserController.java (hunk 1 above)
Comment (merlindk) [reply src/main/java/com/cema/users/controllers/UserController.java:83]: is already included in that method call

File: src/main/java/com/cema/users/controllers/UserController.java (hunk 2 above)
Comment (merlindk) [reply src/main/java/com/cema/users/controllers/UserController.java]: fixed

**Pull request merged**